WordPress/Classes/ViewRelated/Post/Geolocation/PostGeolocationViewController.m

```diff
@@ -135,37 +119,66 @@ - (void)updateLocation
+    NSString *cancelText = NSLocalizedString(@"OK", "");
+    NSString *otherButtonTitle = nil;
+    if (error.domain == kCLErrorDomain && error.code == kCLErrorDenied) {
+        otherButtonTitle = NSLocalizedString(@"Open Settings", @"Go to the settings app");
+        title = NSLocalizedString(@"Location", @"Title for alert when access to the media library is not granted by the user");
+        message = NSLocalizedString(@"WordPress needs permission to access your device location in order to geotag your post. Please change the privacy settings if you wish to allow this.",  @"Explaining to the user why the app needs access to the device location.");
+    }
+    
+    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:title message:message preferredStyle:UIAlertControllerStyleAlert];
+    UIAlertAction *okAction = [UIAlertAction actionWithTitle:cancelText style:UIAlertActionStyleCancel handler:nil];
+    [alertController addAction:okAction];
+    
+    if (otherButtonTitle) {
+        UIAlertAction *otherAction = [UIAlertAction actionWithTitle:otherButtonTitle style:UIAlertActionStyleDefault handler:^(UIAlertAction *action) {
+            NSURL *settingsURL = [NSURL URLWithString:UIApplicationOpenSettingsURLString];
+            [[UIApplication sharedApplication] openURL:settingsURL];
+        }];
+        [alertController addAction:otherAction];
+    }
+    [self presentViewController:alertController animated:YES completion:nil];
+}
+
 - (void)refreshView
 {
-    [self refreshToolbar];
+    [self refreshNavigationBar];
 
-    if ([[LocationService sharedService] locationServiceRunning]) {
+    if ([self.locationService locationServiceRunning]) {
         self.geoView.coordinate = nil;
         self.geoView.address = NSLocalizedString(@"Finding your location...", @"Geo-tagging posts, status message when geolocation is found.");
 
     } else if (self.post.geolocation) {
         self.geoView.coordinate = self.post.geolocation;
-        self.geoView.address = [[LocationService sharedService] lastGeocodedAddress];
+        self.geoView.address = [self.locationService lastGeocodedAddress];
 
     } else {
         self.geoView.coordinate = nil;
-        self.geoView.address = [[LocationService sharedService] lastGeocodedAddress];
+        self.geoView.address = [self.locationService lastGeocodedAddress];
     }
 }
 
-- (void)refreshToolbar
+- (void)refreshNavigationBar
 {
     UIBarButtonItem *leftFixedSpacer = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFixedSpace target:nil action:nil];
     UIBarButtonItem *rightFixedSpacer = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFixedSpace target:nil action:nil];
-    UIBarButtonItem *centerFlexSpacer = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFlexibleSpace target:nil action:nil];
 
     leftFixedSpacer.width = -2.0f;
     rightFixedSpacer.width = -5.0f;
```

Comment:
One of these days.... I swear... I'm going to find all the remaining `f` modifiers and purge them.  Nothing to change here... just ... ranting.